docs/examples/all-options.yaml

```diff
@@ -7,79 +8,103 @@ spec:
     commonName:
       required: true
       value: "example.com"
+      validations:
+        - rule: self.endsWith('.com')
+          message: CommonName must end with '.com'
     dnsNames:
       required: false
       values:
-      - "example.com"
-      - "*.example.com"
+        - "example.com"
+        - "*.example.com"
+      validations:
+        - rule: self.size() =< 24
+          message: DNSName must be no more than 24 characters
     ipAddresses:
-      values:
-      - "1.2.3.4"
-      - "10.0.1.*"
+      required: false
+      values: ["*"]
+      validations:
+        - rule: self.matches('\d+\.\d+\.\d+\.\d+')
+          message: IPAddress must be a valid IPv4 address
     uris:
+      required: false
       values:
-      - "spiffe://example.org/ns/*/sa/*"
+        - "spiffe://example.org/ns/*/sa/*"
+      validations:
+        - rule: self.startsWith('spiffe://%s/ns/%s/sa/'.format(['example.org',cr.namespace]))
+          message: URI must be a valid SPIFFE ID in trust domain bound to request namespace
     emailAddresses:
+      required: false
       values:
-      - "*@example.com"
+        - "*@example.com"
+      validations:
+        - rule: self.size() =< 24
+          message: EmailAddress must be no more than 24 characters
     isCA: false
     usages:
-    - "server auth"
-    - "client auth"
+      - "server auth"
+      - "client auth"
     subject:
       organizations:
-        values: ["hello-world"]
+        required: false
+        values: ["*"]
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
       countries:
+        required: false
         values: ["*"]
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
       organizationalUnits:
+        required: false
         values: ["*"]
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
       localities:
+        required: false
         values: ["*"]
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
       provinces:
+        required: false
         values: ["*"]
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
       streetAddresses:
+        required: false
         values: ["*"]
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
       postalCodes:
+        required: false
         values: ["*"]
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
       serialNumber:
+        required: false
         value: "*"
-
+        validations:
+          - rule: self.size() > 0
+            message: must not be empty
   constraints:
     minDuration: 1h
     maxDuration: 24h
     privateKey:
       algorithm: RSA
       minSize: 2048
       maxSize: 4096
-
+  plugins:
+    rego:
```

Comment:
This plugin is not available in the default distribution.

Reply:
I know, and no plugins are. This policy example is a all-options/max example, so I think we should include set values on all fields available in the API.